tekton/resources/checks/check-pr-has-kind-label.yaml

```diff
@@ -0,0 +1,216 @@
+  verbs: ["get", "list", "watch"]
+- apiGroups: ["tekton.dev"]
+  resources: ["eventlisteners", "triggerbindings", "triggertemplates"]
+  verbs: ["get", "list"]
+- apiGroups: ["tekton.dev"]
+  resources: ["pipelineruns", "pipelineresources", "taskruns"]
+  verbs: ["create"]
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: gh-pr-label-checker
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: gh-pr-label-checker-rolebinding
+subjects:
+- apiGroup: ""
+  kind: ServiceAccount
+  name: gh-pr-label-checker
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: check-labels-trigger
+---
+apiVersion: tekton.dev/v1alpha1
+kind: TriggerBinding
+metadata:
+  name: trigger-to-check-pr-labels
+spec:
+  params:
+  - name: labels
+    value: $(body.pull_request.labels)
+  - name: gitRepository
+    value: $(body.repository.html_url)
+  - name: pullRequestUrl
+    value: $(body.pull_request.html_url)
+---
+apiVersion: tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+  name: github-pull-request-label-checker
+spec:
+  serviceAccountName: gh-pr-label-checker
+  triggers:
+    - name: trigger
+      interceptors:
+        - cel:
+            filter: "header.match('X-GitHub-Event', 'pull_request') && (body.action == 'opened' || ((body.action == 'labeled' || body.action == 'unlabeled') && body.label.name.contains('kind')))"
+      bindings:
+        - name: trigger-to-check-pr-labels
+      template:
+        name: check-pull-request-labels
+---
+apiVersion: tekton.dev/v1alpha1
+kind: TriggerTemplate
+metadata:
+  name: check-pull-request-labels
+spec:
+  params:
+  - name: labels
+    description: List of labels currently on the Pull Request
+    type: string
+  - name: gitRepository
+    type: string
+  - name: pullRequestUrl
+    type: string
+  resourcetemplates:
+  - apiVersion: tekton.dev/v1alpha1
+    kind: TaskRun
+    metadata:
+      name: check-pr-labels-$(uid)
+    spec:
+      taskSpec:
+        inputs:
+          resources:
+          - name: source
+            type: git
+          - name: pr
+            type: pullRequest
+          params:
+          - name: labels 
+            description: The labels attached to the Pull Request
+          - name: gitRepository
+            description: The URL of the git repository
+          - name: pullRequestUrl
+            description: The URL of the Pull Request
+        outputs:
+          resources:
+          - name: pr
+            type: pullRequest
+        volumes:
+        - name: label-config-v2
+          configMap:
+            name: label-config-v2
+        steps:
+        - name: copy-pr-to-output
+          image: busybox
+          script: |
+            #!/bin/sh
+            mkdir -p $(outputs.resources.pr.path)
+            cp -r $(inputs.resources.pr.path)/* $(outputs.resources.pr.path)/
+        - name: install-pyyaml
+          image: python:3-alpine
+          script: |
+            pip install pyyaml --user
```

Comment:
Does this work? 
I guess `pip install pyyaml --user` downloads it into `/home/tekton` which stays for the next container.
I would prefer not to install packages as part of the CI job.

Reply:
Yes this does work!  I am using it parse `labels.yaml` in the next step.  If we don't need it then I will remove it, but it seems like we do.